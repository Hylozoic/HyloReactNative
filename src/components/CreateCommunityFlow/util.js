export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'
