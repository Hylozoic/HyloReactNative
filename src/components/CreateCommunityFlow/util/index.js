export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'

export function formatDomainWithUrl (communityDomain) {
  if (!communityDomain) return
  let formattedDomain = communityDomain.replace(/hylo\.com\/c\/?/, '')
  if (formattedDomain !== '') {
    formattedDomain = 'hylo.com/c/' + formattedDomain
  }
  return formattedDomain
}

export function removeUrlFromDomain (communityDomain) {
  if (!communityDomain) return
  return communityDomain.replace('hylo.com/c/', '')
}
