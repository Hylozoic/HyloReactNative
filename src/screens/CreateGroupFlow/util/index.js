export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'

export function formatDomainWithUrl (groupDomain) {
  if (!groupDomain) return
  let formattedDomain = groupDomain.replace(/hylo\.com\/groups\/?/, '')
  if (formattedDomain !== '') {
    formattedDomain = 'hylo.com/groups/' + formattedDomain
  }
  return formattedDomain
}

export function removeDomainFromURL (groupDomain) {
  if (!groupDomain) return
  return groupDomain.replace('hylo.com/groups/', '')
}
