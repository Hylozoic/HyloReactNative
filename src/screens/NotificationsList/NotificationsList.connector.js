import { connect } from 'react-redux'
import { isEmpty } from 'lodash'
import {
  FETCH_NOTIFICATIONS,
  fetchNotifications,
  getNotifications,
  getHasMoreNotifications,
  markActivityRead,
  markAllActivitiesRead,
  updateNewNotificationCount
} from './NotificationsList.store'
import getMemberships from 'store/selectors/getMemberships'

export function mapStateToProps (state, props) {
  return {
    hasMore: getHasMoreNotifications(state),
    pending: state.pending[FETCH_NOTIFICATIONS],
    notifications: getNotifications(state, props),
    currentUserHasMemberships: !isEmpty(getMemberships(state))
  }
}

export function mapDispatchToProps (dispatch, { navigation }) {
  return {
    fetchNotifications: () => dispatch(fetchNotifications()),
    fetchMore: offset => dispatch(fetchNotifications({ offset })),
    markAllActivitiesRead: () => dispatch(markAllActivitiesRead()),
    markActivityRead: id => dispatch(markActivityRead(id)),
    updateNewNotificationCount: () => dispatch(updateNewNotificationCount()),
    goToCreateGroup: () => navigation.navigate('Create Group')
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
