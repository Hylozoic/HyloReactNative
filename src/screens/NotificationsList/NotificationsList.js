import React, { useEffect, useState, useCallback } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { FlatList, TouchableOpacity, View, Text } from 'react-native'
import { useIsFocused, useNavigation } from '@react-navigation/native'
import { useTranslation } from 'react-i18next'
import ModalHeader from 'navigation/headers/ModalHeader'
import NotificationCard from 'components/NotificationCard'
import CreateGroupNotice from 'components/CreateGroupNotice'
import Loading from 'components/Loading'
import cardStyles from 'components/NotificationCard/NotificationCard.styles'
import styles from './NotificationsList.styles'
import {
  FETCH_NOTIFICATIONS,
  fetchNotifications,
  getNotifications,
  getHasMoreNotifications,
  markActivityRead,
  markAllActivitiesRead,
  updateNewNotificationCount
} from './NotificationsList.store'
import getMemberships from 'store/selectors/getMemberships'
import { isEmpty } from 'lodash'

const NOTIFICATIONS_PAGE_SIZE = 20

export const NotificationsList = (props) => {
  const navigation = useNavigation()
  const dispatch = useDispatch()
  const isFocused = useIsFocused()
  const { t } = useTranslation()

  const [ready, setReady] = useState(false)

  const hasMore = useSelector(getHasMoreNotifications)
  const pending = useSelector(state => state.pending[FETCH_NOTIFICATIONS])
  const notifications = useSelector(state => getNotifications(state, props))
  const currentUserHasMemberships = useSelector(state => !isEmpty(getMemberships(state)))

  const fetchNotificationsHandler = useCallback(() => {
    dispatch(fetchNotifications(NOTIFICATIONS_PAGE_SIZE))
  }, [dispatch])

  const fetchMoreHandler = useCallback((offset) => {
    dispatch(fetchNotifications(NOTIFICATIONS_PAGE_SIZE, offset))
  }, [dispatch])

  const markAllActivitiesReadHandler = useCallback(() => {
    dispatch(markAllActivitiesRead())
  }, [dispatch])

  const markActivityReadHandler = useCallback((id) => {
    dispatch(markActivityRead(id))
  }, [dispatch])

  const updateNewNotificationCountHandler = useCallback(() => {
    dispatch(updateNewNotificationCount())
  }, [dispatch])

  const goToCreateGroupHandler = useCallback(() => {
    navigation.navigate('Create Group')
  }, [navigation])

  const setHeader = useCallback(() => {
    navigation.setOptions({
      title: t('Notifications'),
      header: props => (
        <ModalHeader
          {...props}
          headerRightButtonLabel={t('Mark as read')}
          headerRightButtonOnPress={markAllActivitiesReadHandler}
        />
      )
    })
  }, [navigation, t, markAllActivitiesReadHandler])

  useEffect(() => {
    setHeader()
    fetchNotificationsHandler()
    updateNewNotificationCountHandler()
  }, [setHeader, fetchNotificationsHandler, updateNewNotificationCountHandler])

  useEffect(() => {
    if (isFocused) {
      setHeader()
      fetchNotificationsHandler()
      updateNewNotificationCountHandler()
    }
  }, [isFocused, setHeader, fetchNotificationsHandler, updateNewNotificationCountHandler])

  useEffect(() => {
    if (!pending) {
      setReady(true)
    }
  }, [pending])

  if (!currentUserHasMemberships) {
    return (
      <CreateGroupNotice
        goToCreateGroup={goToCreateGroupHandler}
        text={t('No notifications here, try creating your own Group!')}
      />
    )
  }

  if (ready && !pending && notifications.length === 0) {
    return <Text style={styles.center}>{t('Nothing new for you!')}</Text>
  }

  return (
    <View style={styles.notificationsList}>
      {pending && (
        <View style={cardStyles.container}>
          <View style={cardStyles.content}>
            <Loading />
          </View>
        </View>
      )}
      <FlatList
        data={notifications}
        keyExtractor={item => item.id}
        onEndReached={hasMore ? () => fetchMoreHandler(notifications.length) : null}
        renderItem={({ item }) => (
          <NotificationRow
            markActivityRead={markActivityReadHandler}
            notification={item}
          />
        )}
      />
    </View>
  )
}

export const NotificationRow = ({ markActivityRead, notification }) => {
  return (
    <View>
      <TouchableOpacity
        onPress={() => {
          if (notification.unread) markActivityRead(notification.activityId)
          notification.onPress()
        }}
      >
        <NotificationCard notification={notification} />
      </TouchableOpacity>
    </View>
  )
}

export default NotificationsList
