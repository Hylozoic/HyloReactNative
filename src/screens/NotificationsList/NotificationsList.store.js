import { createSelector } from 'reselect'
import { createSelector as ormCreateSelector } from 'redux-orm'
import { get, find, pick } from 'lodash/fp'
import { TextHelpers } from 'hylo-shared'
import orm from 'store/models'
import { makeGetQueryResults } from 'store/reducers/queryResults'
import postFieldsFragment from 'graphql/fragments/postFieldsFragment'
import { modalScreenName } from 'navigation/linking/helpers'

import {
  ACTION_NEW_COMMENT,
  ACTION_TOPIC,
  ACTION_JOIN_REQUEST,
  ACTION_APPROVED_JOIN_REQUEST,
  ACTION_MENTION,
  ACTION_COMMENT_MENTION,
  ACTION_ANNOUNCEMENT,
  NOTIFICATIONS_WHITELIST
} from 'store/models/Notification'

export const MODULE_NAME = 'NotificationsList'
export const FETCH_NOTIFICATIONS = `${MODULE_NAME}/FETCH_NOTIFICATIONS`
export const MARK_ACTIVITY_READ = `${MODULE_NAME}/MARK_ACTIVITY_READ`
export const MARK_ALL_ACTIVITIES_READ = `${MODULE_NAME}/MARK_ALL_ACTIVITIES_READ`
export const UPDATE_NEW_NOTIFICATION_COUNT = `${MODULE_NAME}/UPDATE_NEW_NOTIFICATION_COUNT`
export const UPDATE_NEW_NOTIFICATION_COUNT_PENDING = `${UPDATE_NEW_NOTIFICATION_COUNT}_PENDING`

export const NOTIFICATION_TEXT_MAX = 76

export function fetchNotifications (first = 20, offset = 0) {
  return {
    type: FETCH_NOTIFICATIONS,
    graphql: {
      query: `query ($first: Int, $offset: Int) {
        notifications (first: $first, offset: $offset, order: "desc") {
          total
          hasMore
          items {
            id
            createdAt
            activity {
              id
              actor {
                id
                name
                avatarUrl
              }
              comment {
                id
                text
              }
              post {
                ${postFieldsFragment(true)}
              }
              group {
                id
                name
                slug
              }
              meta {
                reasons
              }
              action
              unread
            }
          }
        }
      }`,
      variables: { first, offset }
    },
    meta: {
      afterInteractions: true,
      extractModel: 'Notification',
      extractQueryResults: {
        getItems: get('payload.data.notifications')
      },
      resetCount: true
    }
  }
}

export function markActivityRead (id) {
  return {
    type: MARK_ACTIVITY_READ,
    graphql: {
      query: `mutation ($id: ID) {
        markActivityRead(id: $id) {
          id
        }
      }`,
      variables: { id }
    },
    meta: {
      id,
      optimistic: true
    }
  }
}

export function markAllActivitiesRead () {
  return {
    type: MARK_ALL_ACTIVITIES_READ,
    graphql: {
      query: `mutation {
        markAllActivitiesRead {
          success
        }
      }`
    },
    meta: {
      optimistic: true
    }
  }
}

export function updateNewNotificationCount () {
  return {
    type: UPDATE_NEW_NOTIFICATION_COUNT,
    graphql: {
      query: `mutation ($changes: MeInput) {
        updateMe(changes: $changes) {
          id
        }
      }`,
      variables: {
        changes: {
          newNotificationCount: 0
        }
      }
    },
    meta: {
      optimistic: true
    }
  }
}

export const truncateHTML = html => TextHelpers.truncateText(TextHelpers.htmlToText(html).replace(/\n/g, ' '), NOTIFICATION_TEXT_MAX)

export const truncateText = text => TextHelpers.truncateText(text, NOTIFICATION_TEXT_MAX)

export function refineActivity ({ action, actor, comment, group, post, meta }, { navigate }) {
  switch (action) {
    case ACTION_COMMENT_MENTION:
      return {
        body: `wrote: ${truncateHTML(comment.text)}`,
        header: 'mentioned you in a comment on',
        nameInHeader: true,
        onPress: () => navigate(modalScreenName('Post Details'), { id: post.id }),
        title: post.title
      }

    case ACTION_NEW_COMMENT:
      return {
        body: `wrote: ${truncateHTML(comment.text)}`,
        header: 'New Comment on',
        onPress: () => navigate(modalScreenName('Post Details'), { id: post.id }),
        title: post.title
      }

    case ACTION_MENTION:
      return {
        body: `wrote: ${truncateHTML(post.details)}`,
        header: 'mentioned you',
        onPress: () => navigate(modalScreenName('Post Details'), { id: post.id }),
        nameInHeader: true
      }

    case ACTION_TOPIC: {
      const topicReason = find(r => r.startsWith('tag: '), meta.reasons)
      const topic = topicReason.split(': ')[1]
      return {
        body: `wrote: ${truncateHTML(post.details)}`,
        header: 'New Post in',
        onPress: () => navigate(modalScreenName('Post Details'), { id: post.id }),
        topic
      }
    }

    case ACTION_JOIN_REQUEST:
      return {
        body: 'asked to join',
        group: group.name,
        header: 'New join request',
        nameInHeader: true,
        onPress: () => navigate('Group Settings', {
          screen: 'Join Requests',
          params: {
            groupId: group.id, groupSlug: group.slug
          }
        })
      }

    case ACTION_APPROVED_JOIN_REQUEST:
      return {
        body: 'approved your request to join',
        group: group.name,
        header: 'Join Request Approved',
        onPress: () => navigate('Feed', { groupId: group.id })
      }
    case ACTION_ANNOUNCEMENT:
      return {
        body: `wrote: ${truncateText(post.title)}`,
        header: 'posted an announcement',
        onPress: () => navigate(modalScreenName('Post Details'), { id: post.id }),
        nameInHeader: true
      }
  }
}

export function refineNotification (navigation) {
  return ({ activity, createdAt, id }, i, notifications) => {
    const { actor, meta, unread } = activity
    // Only show separator between read and unread notifications

    const avatarSeparator = i !== notifications.length - 1
      ? unread !== notifications[i + 1].activity.unread
      : false

    return {
      id,
      activityId: activity.id,
      actor: pick(['avatarUrl', 'name'], actor),
      avatarSeparator,
      createdAt: TextHelpers.humanDate(createdAt),
      ...refineActivity(activity, navigation),
      unread,
      reasons: meta.reasons
    }
  }
}

const getNotificationsResults = makeGetQueryResults(FETCH_NOTIFICATIONS)

export const getHasMoreNotifications = createSelector(
  getNotificationsResults,
  get('hasMore')
)

export const getNotifications = ormCreateSelector(
  orm,
  (_, { navigation }) => navigation,
  (session, navigation) => session.Notification
    .all()
    .orderBy(m => Number(m.id), 'desc')
    .toModelArray()
    .map(refineNotification(navigation))
    .filter(n => n.reasons.every(r => reasonInWhitelist(r, NOTIFICATIONS_WHITELIST)))
)

export function reasonInWhitelist (reason, whitelist) {
  const reasonSubstring = reason.indexOf(':') === -1 ? reason : reason.substring(0, reason.indexOf(':'))
  const response = whitelist.indexOf(reasonSubstring) !== -1
  return response
}
