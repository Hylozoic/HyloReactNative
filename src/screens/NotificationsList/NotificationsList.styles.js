import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  notificationsList: {
    backgroundColor: 'white',
    position: 'relative'
  },
  loadingContainer: {
    position: 'absolute',
    left: 0,
    top: 12,
    zIndex: 2
  },
  center: {
    padding: 20
  }
})
