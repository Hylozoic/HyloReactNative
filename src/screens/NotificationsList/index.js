import component from './NotificationsList'

export default component
