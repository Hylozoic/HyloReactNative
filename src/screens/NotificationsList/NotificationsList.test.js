import { FlatList, TouchableOpacity } from 'react-native'
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import TestRenderer from 'react-test-renderer'
import { simulate } from 'util/testing'
import Loading from 'components/Loading'
import { NotificationsList, NotificationRow } from './NotificationsList'

jest.mock('util/platform', () => ({ isIOS: true }))

jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (str) => str }
    return Component
  },
  useTranslation: (domain) => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {})
      }
    }
  }
}))

describe('NotificationsList', () => {
  let props = null
  let shallowRenderer = null

  beforeEach(() => {
    shallowRenderer = new ShallowRenderer()
    props = {
      fetchMore: jest.fn(),
      fetchNotifications: () => {},
      t: str => str,
      updateNewNotificationCount: () => {},
      hasMore: true,
      isFocused: true,
      markActivityRead: jest.fn(),
      markAllRead: jest.fn(),
      pending: false,
      currentUserHasMemberships: true,
      notifications: [
        {
          activityId: '1',
          actor: { avatarUrl: 'https://wombat.com', name: 'Wombat McAardvark' },
          avatarSeparator: false,
          createdAt: '2 min ago',
          id: '1',
          onPress: jest.fn(),
          unread: false
        }
      ],
      setRightButton: () => {},
      navigation: {
        setOptions: jest.fn()
      }
    }
  })

  it('matches the last snapshot', () => {
    const renderer = TestRenderer.create(<NotificationsList {...props} />)
    renderer.getInstance().setState({ ready: true })
    expect(renderer.toJSON()).toMatchSnapshot()
  })

  it('Shows Loading when pending is true', () => {
    props.pending = true
    props.notifications = []
    shallowRenderer.render(<NotificationsList {...props} />)
    const actual = shallowRenderer.getRenderOutput()

    const loadingComponent = actual.props.children[0].props.children // Accessing the Loading component within the structure
    expect(loadingComponent).toEqual(<Loading />)
  })

  it('returns a message when no notifications are available', () => {
    props.notifications = []
    const renderer = TestRenderer.create(<NotificationsList {...props} />)
    renderer.getInstance().setState({ ready: true })
    const hasMessage = renderer.toJSON().children.find(c => c.includes('Nothing new'))
    expect(hasMessage).not.toBeFalsy()
  })

  it('marks the notification read on notification touch', () => {
    const notification = props.notifications[0]
    notification.unread = true
    const root = TestRenderer.create(<NotificationsList {...props} />).root
    root.instance.setState({ ready: true })
    simulate(root.findAllByType(TouchableOpacity)[0], 'press')
    expect(props.markActivityRead).toHaveBeenCalledWith(notification.activityId)
  })

  it('calls notification.onPress on notification touch', () => {
    const root = TestRenderer.create(<NotificationsList {...props} />).root
    root.instance.setState({ ready: true })
    const notification = root.props.notifications[0]
    simulate(root.findAllByType(TouchableOpacity)[0], 'press')
    expect(notification.onPress).toHaveBeenCalled()
  })

  it('calls fetchMore when scrolled to end of list', () => {
    const root = TestRenderer.create(<NotificationsList {...props} />).root
    root.instance.setState({ ready: true })
    simulate(root.findByType(FlatList), 'endReached')
    expect(props.fetchMore).toHaveBeenCalledWith(props.notifications.length)
  })

  it('matches the last snapshot for NotificationRow', () => {
    shallowRenderer.render(<NotificationRow
      markActivityRead={props.markActivityRead}
      notification={props.notifications[0]}
                           />)
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })

  it('matches the renders CreateGroupNotice if user does not have memberships', () => {
    props.currentUserHasMemberships = false
    const renderer = TestRenderer.create(<NotificationsList {...props} />)
    renderer.getInstance().setState({ ready: true })
    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
