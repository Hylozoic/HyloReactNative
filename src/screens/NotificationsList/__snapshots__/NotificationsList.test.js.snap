// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsList matches the last snapshot 1`] = `
<View>
  <RCTScrollView
    data={
      [
        {
          "activityId": "1",
          "actor": {
            "avatarUrl": "https://wombat.com",
            "name": "Wombat McAardvark",
          },
          "avatarSeparator": false,
          "createdAt": "2 min ago",
          "id": "1",
          "onPress": [MockFunction],
          "unread": false,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View>
              <View>
                <View>
                  <View
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        undefined,
                      ]
                    }
                  >
                    <FastImageView
                      defaultSource={null}
                      resizeMode="cover"
                      source={
                        {
                          "uri": "https://wombat.com",
                        }
                      }
                      style={
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View>
                <Text
                  numberOfLines={2}
                >
                  <Text />
                </Text>
                <Text>
                  <Text>
                    Wombat 
                  </Text>
                </Text>
                <Text>
                  2 min ago
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`NotificationsList matches the last snapshot for NotificationRow 1`] = `
<View>
  <TouchableOpacity
    onPress={[Function]}
  >
    <NotificationCard
      notification={
        {
          "activityId": "1",
          "actor": {
            "avatarUrl": "https://wombat.com",
            "name": "Wombat McAardvark",
          },
          "avatarSeparator": false,
          "createdAt": "2 min ago",
          "id": "1",
          "onPress": [MockFunction],
          "unread": false,
        }
      }
    />
  </TouchableOpacity>
</View>
`;

exports[`NotificationsList matches the renders CreateGroupNotice if user does not have memberships 1`] = `
<View>
  <Text>
    No notifications here, try creating your own Group!
  </Text>
  <Image
    source={
      {
        "process": [Function],
      }
    }
  />
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View>
        <View>
          <Text>
            Start a Group
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
